dCommon/AMFDeserialize.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "BitStream.h"
+
+#include <vector>
+#include <string>
+
+class AMFArrayValue;
+class AMFDeserialize {
+    public:
+        /**
+         * Read an AMF3 value from a bitstream.
+         * 
+         * @param inStream inStream to read value from.
+         * @param doFullProcess currently this is here for arrays but should be removed and have its work be done in the AMFArray switch case.
```

Comment:
The parameter `doFullProcess` does not exist; should that be `readStartOfArray`?